### PR TITLE
[SUBTASK] [KYUUBI #3420] Spark SQL Engine register zk with ui info

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

See more in #3420 
This pr aim to improve spark sql engine register zookeeper with spark ui info

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zwangsheng** [comment]
cc @pan3793 @yaooqinn

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/ZookeeperSparkEngineRegisterSuite.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.apache.kyuubi.ha.client.zookeeper.ZookeeperClientProvider
+
+class ZookeeperSparkEngineRegisterSuite extends WithDiscoverySparkSQLEngine
+  with WithEmbeddedZookeeper {
+  override def withKyuubiConf: Map[String, String] =
+    super.withKyuubiConf ++ zookeeperConf ++ Map(
+      PROXY_SPARK_UI_ENABLED.key -> "true",
+      "spark.ui.enabled" -> "true")
+
+  override val namespace: String = s"/kyuubi/deregister_test/${UUID.randomUUID().toString}"
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/ZookeeperSparkEngineRegisterSuite.scala:33]:
```suggestion
  override val namespace: String = s"/kyuubi/deregister_test/${UUID.randomUUID}"
```

File: kyuubi-common/src/main/scala/org/apache/kyuubi/service/FrontendService.scala [hunk 2]
```diff
@@ -41,4 +41,9 @@ trait FrontendService {
    * An optional `ServiceDiscovery` for [[FrontendService]] to expose itself
    */
   val discoveryService: Option[Service]
+
+  /**
+   * An optional `UI`for each engine expose it's ui
+   */
+  def ui: Option[String] = None
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/service/FrontendService.scala:48]: How about changing it to 
```
def extraConfs: Map[String, String] = Map.empty
```
?
We may want to expose more information in the future, e.g. metrics

cc @turboFei @yaooqinn @cxzl25 @wForget

Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/service/FrontendService.scala:48]: +1 for 
```
def extraConfs: Map[String, String] = Map.empty
```

Comment (zwangsheng) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/service/FrontendService.scala]: Thanks for suggestion, it has been revised.

**Pull request closed without merging**